### PR TITLE
[6165] XB buttons are not role aware

### PR DESCRIPTION
**Pull request by jvega190**
https://github.com/craftercms/craftercms/issues/6165

### CODE REVIEW

File: ui/app/src/components/PreviewConcierge/PreviewConcierge.tsx [hunk 1]
```diff
@@ -609,7 +620,8 @@ export function PreviewConcierge(props: PropsWithChildren<{}>) {
             contentTypes,
             requestedSourceMapPaths,
             dispatch,
-            completeAction: fetchPrimaryGuestModelComplete
+            completeAction: fetchPrimaryGuestModelComplete,
+            permissions
```
Comment (rart) [inline ui/app/src/components/PreviewConcierge/PreviewConcierge.tsx:624]: This needs to be added to the upToDateRefs. Right now its stale. You should be getting compilation warnings.

File: ui/guest/src/react/ZoneMenu.tsx [hunk 2]
```diff
@@ -146,15 +156,17 @@ export function ZoneMenu(props: ZoneMenuProps) {
       const maxValidation = validations?.maxCount?.value;
       const minValidation = validations?.minCount?.value;
       const trashableValidation = minValidation ? minValidation < numOfItemsInContainerCollection : true;
-      const duplicateValidation = maxValidation ? maxValidation > numOfItemsInContainerCollection : true;
+      const duplicateValidation =
+        permissions.includes('content_create') &&
```
Comment (rart) [inline ui/guest/src/react/ZoneMenu.tsx:160]: This is to delete. Isn't there a delete permission?

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/guest/src/react/ZoneMenu.tsx (hunk 2 above)
Comment (jvega190) [reply ui/guest/src/react/ZoneMenu.tsx:160]: Per conversation this section is for duplicateValidation, not for trashableValidation

File: ui/app/src/components/PreviewConcierge/PreviewConcierge.tsx [hunk 3]
```diff
@@ -298,6 +308,7 @@ export function PreviewConcierge(props: PropsWithChildren<{}>) {
   const store = useStore<GlobalState>();
   const { id: siteId, uuid } = useActiveSite() ?? {};
   const user = useActiveUser();
+  const permissions = user?.permissionsBySite[siteId];
```
Comment (rart) [inline ui/app/src/components/PreviewConcierge/PreviewConcierge.tsx]: Actually, this is part of the user object which is already in the updated refs. There's no point on having this declared outside of the effect, is there?

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/app/src/components/PreviewConcierge/PreviewConcierge.tsx (hunk 3 above)
Comment (jvega190) [reply ui/app/src/components/PreviewConcierge/PreviewConcierge.tsx]: no point, changing it to use user from refs to get permissions

**Review by rart — APPROVED** [review]

**Pull request merged**